src/Test/FixturesTrait.php

```diff
@@ -45,15 +45,15 @@ protected function getContainer(): ContainerInterface
             $options = [
                 'environment' => $environment,
             ];
-            $kernel = $this->createKernel($options);
-            $kernel->boot();
-
-            $container = $kernel->getContainer();
-            if ($container->has('test.service_container')) {
-                $this->containers[$environment] = $container->get('test.service_container');
-            } else {
-                $this->containers[$environment] = $container;
+
+            // Check that the kernel has not been booted separately (eg. with static::createClient())
+            $kernel = static::$kernel;
+            if (null === $kernel) {
+                $kernel = $this->bootKernel($options);
             }
+
+            // bootKernel() has logic to ensure static::$container is the correct container
+            $this->containers[$environment] = static::$container;
```

Comment:
Looks like the `static::$container` property was introduced in Symfony 4, which is causing the build to fail for Symfony 3.4. Should I undo this change?
```suggestion
            $container = $kernel->getContainer();
            if ($container->has('test.service_container')) {
                $this->containers[$environment] = $container->get('test.service_container');
            } else {
                $this->containers[$environment] = $container;
            }
```

Reply:
How about `$this->containers[$environment] = static::$container ?? $kernel->getContainer();`

Reply:
This still won't work on Symfony 3.4 I don't think. Because it will get the "real" container instead of the test one. eg. this is how KernelTestCase does it in Symfony 4:

```php
self::$kernelContainer = $container = static::$kernel->getContainer();
static::$container = $container->has('test.service_container') ? $container->get('test.service_container') : $container;
```

I think I'll just revert the change.